pennylane/templates/broadcast.py

```diff
@@ -35,21 +37,20 @@ def wires_ring(wires):
     if len(wires) in [0, 1]:
         return []
     elif len(wires) == 2:
-        # exception: for 2 wires ring is set equal to chain,
-        # to avoid duplication of gate
-        return [[wires[0], wires[1]]]
+        # deviation from the rule: for 2 wires ring is set equal to chain,
+        # to avoid duplication of single gate
+        return [wires.subset([0, 1])]
     else:
-        ring = list(wires) + list(wires[0:1])
-        sequence = [[ring[i], ring[i + 1]] for i in range(len(wires))]
+        sequence = [wires.subset([i, i + 1], periodic_boundary=True) for i in range(len(wires))]
         return sequence
 
 
 def wires_pyramid(wires):
     """Wire sequence for the pyramid pattern."""
     sequence = []
     for layer in range(len(wires) // 2):
-        temp = wires[layer : len(wires) - layer]
-        sequence += [[temp[i], temp[i + 1]] for i in range(0, len(temp) - 1, 2)]
+        subset = wires[layer : len(wires) - layer]
+        sequence += [subset.subset([i, i + 1]) for i in range(0, len(subset) - 1, 2)]
```

Comment:
Hmm, somehow taken out of context it can appear slightly confusing what exactly the `subset` object stands for here and what `subset.subset` would mean later. Perhaps `block` or `group` or so could help out? :sweat_smile: not sure how to best refer to it